Lib/test/support/__init__.py

```diff
@@ -240,8 +240,10 @@ def _save_and_remove_module(name, orig_modules):
     """
     # try to import the module and raise an error if it can't be imported
     if name not in sys.modules:
+        orig_modules = sys.modules.copy()
         __import__(name)
-        del sys.modules[name]
+        sys.modules.clear()
+        sys.modules.update(orig_modules)
```

Comment:
I considered this option, but some modules can not support re-importing. So it is safer to restore only explicitly mentioned modules and their submodules.
See #28654.

Reply:
Oh, sorry, Serhiy. I missed your PR :(